ocaml/xapi/xapi_http.ml

```diff
@@ -44,6 +44,18 @@ open Http
 
 exception NoAuth
 
+let task_id_of_req (req: Http.Request.t) =
+  let all = Http.Request.(req.cookie @ req.query) in
+  if List.mem_assoc "task_id" all
+  then Some (Ref.of_string (List.assoc "task_id" all))
+  else None
+
+let subtask_of_req (req: Http.Request.t) =
```

Comment:
This function-name is a bit confusing.
The meaning is really "subtask_of of req", which means the req is (or corresponds to) a subtask of the task which this function returns.
"subtask_of_of_req" would be a confusing name too, though.
So would "subtaskof_of_req" even though it might be slightly clearer.
How about "parent_task_of_req" or perhaps better still, do something like the code below?
```
let ref_param_of_req req param_name =
  let all = Http.Request.(req.cookie @ req.query) in
  try Some (Ref.of_string (List.assoc "subtask_of" all))
  with Not_found -> None
```
then use that function in
`let subtask_of = ref_param_of_req req "subtask_of" in` and
`let task_id = ref_param_of_req req "task_id" in`